### PR TITLE
Perfdata: Add support for more UoMs

### LINKED ISSUE #71: Support for more UoMs in perfdata
## Is your feature request related to a problem? Please describe.
Check plugins currently have only a limited set of UoMs (Units of measurement) available which Icinga understands.

## Describe the solution you'd like
With Icinga v2.13 support for more is added. We have to support them as well.

## Additional context
https://github.com/Icinga/icinga2/issues/7225

### PR DESCRIPTION
**Pull request by Yonas-net**
resolves #71

### CODE REVIEW

**Comment by Yonas-net** [comment]
Depends on [#4369](https://github.com/Icinga/icingaweb2/pull/4369)

File: library/Icingadb/Util/PerfData.php
Comment (nilmerg) [inline library/Icingadb/Util/PerfData.php:456]: I'm not sure if values need to be converted here still. If we expect the input data as being normalized, there's no need to. Also, doesn't the formatting in method `format()` expect the normalized values from Icinga?

File: library/Icingadb/Util/PerfData.php
Comment (nilmerg) [inline library/Icingadb/Util/PerfData.php:704]: The `strtoupper` call isn't necessary anymore.

File: library/Icingadb/Util/PerfData.php
Comment (nilmerg) [inline library/Icingadb/Util/PerfData.php:643]: * Consider to use a `switch (true) { case $this->isPercentage(): ... }` here
  * There are now so much similar control structures, it makes sense to condense them
* Also please put a space between casts and variables everywhere (i.e. `(string) $value`)

File: library/Icingadb/Util/Format.php
Comment (nilmerg) [inline library/Icingadb/Util/Format.php:7]: Please name it `PerfDataFormat` and remove all unused methods. (e.g. `secondsByMonth`)

**Review by nilmerg — CHANGES_REQUESTED** [review]

File: library/Icingadb/Util/PerfData.php
Comment (Yonas-net) [reply library/Icingadb/Util/PerfData.php:456]: Ohh, you're right. I wonder if this is also the case for the thresholds, so are the thresholds normalized as well or does this only affect the plugin output?

File: library/Icingadb/Util/PerfData.php
Comment (Yonas-net) [reply library/Icingadb/Util/PerfData.php:246]: As I understand it, they are different units. For example these units ``mWm, uWm`` are normalized to ``Wm`` and not to ``Wh``. This also affects the different ampere units``Ah, Am, As``. But I might be wrong.

File: library/Icingadb/Util/PerfData.php
Comment (nilmerg) [reply library/Icingadb/Util/PerfData.php:456]: That is a question for @Al2Klimov :wink:

File: library/Icingadb/Util/PerfData.php
Comment (nilmerg) [reply library/Icingadb/Util/PerfData.php:246]: the [table](https://github.com/Icinga/icinga2/pull/8759/files#diff-b5d3095868e71d7e5d627a517bab1e8e40198c49d0536f43e36b60ede5488278R677-R679) shows that for these units the metric backends get *watt-hours*, so it should be `Wh` for them if normalized.

File: library/Icingadb/Util/PerfData.php
Comment (Al2Klimov) [reply library/Icingadb/Util/PerfData.php:456]: Reading about _plugin output normalization_ the first time. 😉

And – regarding the thresholds:

https://github.com/Icinga/icinga2/blob/63b53bfda92930b2332035860082dbcaa5cff9e7/test/icinga-perfdata.cpp#L46-L53

Comment (Yonas-net) [reply library/Icingadb/Util/PerfData.php:456]: > Reading about _plugin output normalization_ the first time. 😉

My mistake.

File: library/Icingadb/Util/PerfDataFormat.php
Comment (nilmerg) [inline library/Icingadb/Util/PerfDataFormat.php:3]: Please change this to `GPLv2` (Remove the plus). I know, the other PerfData files also use it, but it's wrong as well.

File: library/Icingadb/Util/PerfDataFormat.php
Comment (nilmerg) [inline library/Icingadb/Util/PerfDataFormat.php:107]: I'd say, after many weeks and months and years: No.

Don't keep the comment, remove it :wink:

**Review by nilmerg — CHANGES_REQUESTED** [review]
Nice! Two minor changes left and it's ready!

**Review by nilmerg — DISMISSED** [review]
LGTM! Can be merged once tested with actual normalized data. (Blocked by ~~https://github.com/Icinga/icinga2/pull/8761~~ https://github.com/Icinga/icingadb/pull/308 in the meantime)

**Review by nilmerg — APPROVED** [review]

**Pull request merged**
